egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,299 @@
+    parser.add_argument("output_dir",
+                        help="Output data directory")
+
+    print(' '.join(sys.argv))
+
+    args = parser.parse_args()
+    args = CheckArgs(args)
+
+    return args
+
+def CheckArgs(args):
+    if not os.path.exists(args.output_dir):
+        os.makedirs(args.output_dir)
+
+    ## Check arguments.
+    if args.rir_list_file is None:
+        raise Exception("Rir information file must be provided")
+    
+    if not os.path.isfile(args.rir_list_file):
+        raise Exception(args.rir_list_file + "not found")
+    
+    if args.noise_list_file is not None:
+        if not os.path.isfile(args.noise_list_file):
+            raise Exception(args.noise_list_file + "not found")
+
+    if args.num_replica > 1 and args.prefix is None:
+        args.prefix = "rvb"
+        warnings.warn("--prefix is set to 'rvb' as --num-replications is larger than 1.")
+
+    return args
+
+def ParseFileToDict(file, assert2fields = False, value_processor = None):
+    if value_processor is None:
+        value_processor = lambda x: x[0]
+
+    dict = {}
+    for line in open(file, 'r'):
+        parts = line.split()
+        if assert2fields:
+            assert(len(parts) == 2)
+
+        dict[parts[0]] = value_processor(parts[1:])
+    return dict
+
+
+# This is the major function to generate pipeline command for the corruption
+# The rir list would have the following format:
+# --rir-id <string,compulsary> --room-id <string,compulsary> --receiver-position-id <string,optional> --source-position-id <string,optional> --rt-60 < <float,optional> --drr <float, optional> < location(support Kaldi IO strings) >
+# The noise list would have the following format:
+# --noise-id <string,compulsary> --noise-type <choices = (isotropic, point source),compulsary> --bg-fg-type <choices=(background|foreground), default=background> --rir-file <str, compulsary if isotropic, should not be specified if point-source> < location=(support Kaldi IO strings) >
+def CorruptWav(wav_scp, durations, output_dir, room_list, noise_list, snr_string, num_replica, prefix, speech_rvb_probability, noise_adding_probability, max_noises_added):
+    rooms = list_cyclic_iterator(room_list, random_seed = 1)
+    noises = None
+    if len(noise_list) > 0:
+        noises = list_cyclic_iterator(noise_list, random_seed = 1)
+    snrs = list_cyclic_iterator(snr_string.split(':'))
+    command_list = []
+    for i in range(num_replica):
+        keys = wav_scp.keys()
+        keys.sort()
+        for wav_id in keys:
+            wav_pipe = wav_scp[wav_id]
+            wav_dur = durations[wav_id]
+            if prefix is not None:
+                wav_id = prefix + str(i) + "_" + wav_id
+            command = "{0} {1} wav-reverberate".format(wav_id, wav_pipe)
+
+            # pick the room
+            room = rooms.next()
+            command_opts = ""
+            noises_added = []
+            snrs_added = []
+            start_times_added = []
```

Comment:
Background noise is like stationary noise or like stationary noise. It is
long and has the same characteristics for a long duration. It can be noise
like background AC noise or even like a siren or bird sound or perhaps even
music. The foreground noise would be something of short duration like door
taps, keyboard sound, bells, footsteps, dog bark, mic movements etc.
Duration would be the critical difference. Or to be more specific, the
noise energy is concentrated over a short duration for a "foreground"
noise. The foreground noise files I have are not necessarily short files.
They may have the foreground noise repeated after a period of silence (e.g.
door taps separated by 1s of silence). It should ideally not have such
silences, but it might not be possible to get well conditioned noises.


On Thu, Apr 21, 2016 at 3:34 PM Vijayaditya Peddinti <
notifications@github.com> wrote:

> In egs/wsj/s5/steps/data/reverberate_data_dir.py
> <https://github.com/kaldi-asr/kaldi/pull/706#discussion_r60642044>:
>
> > +    for i in range(num_replica):
> > +        keys = wav_scp.keys()
> > +        keys.sort()
> > +        for wav_id in keys:
> > +            wav_pipe = wav_scp[wav_id]
> > +            wav_dur = durations[wav_id]
> > +            if prefix is not None:
> > +                wav_id = prefix + str(i) + "_" + wav_id
> > +            command = "{0} {1} wav-reverberate".format(wav_id, wav_pipe)
> > +
> > +            # pick the room
> > +            room = rooms.next()
> > +            command_opts = ""
> > +            noises_added = []
> > +            snrs_added = []
> > +            start_times_added = []
>
> @vimalmanohar <https://github.com/vimalmanohar> Could you give examples
> of the point source noises you have access to. If it is things like door
> bangs I don't think we should be repeating them. If it is things like
> keyboard sounds it makes sense to repeat them. I am not aware of the
> complete list of point source noises you plan to use, it might be better to
> decide based on that.
>
> For the sake of ensuring all of us are on the same page could you give
> some examples which explain the critical differences between these noises
> in your mind. It would be better if you could give some examples where
> people might have confusion classifying the noises.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/706/files/356fa8258d848e3b092c5b894f6a7b4cab63b29c#r60642044>
>
-- 
Vimal Manohar
PhD Student
Electrical & Computer Engineering
Johns Hopkins University